### ISSUE TITLE
Private projects are indexed by Google

### ISSUE THREAD

**Issue by sneljeroen** [stackblitz/bolt.new#1597]
### Describe the bug

Currently private projects are being indexed by Google, this should not be happening. Each private project should be prompted with a login window before showing the content of the project.

### Link to the Bolt URL that caused the error

https://www.google.com/search?q=site%3Abolt.new&rlz=1C1CHBF_nlNL910NL910&oq=site%3Abolt.new&gs_lcrp=EgZjaHJvbWUyBggAEEUYOTIGCAEQRRg60gEINTU3NGowajeoAgiwAgE&sourceid=chrome&ie=UTF-8

### Steps to reproduce

1. Create a project
2. Set to private
3. When URL is found by Google>
4. Private project is indexed and accessible by anyone using Google.

### Expected behavior

Private projects should prompt a login window without ever showing the contents of the project before authenticating the user.

### Platform

- OS: Windows
- Browser: Chrome
- Version: -

**Comment by sneljeroen** [comment]
With a simple "site:bolt.new" search within Google now private projects are accessible to the Public. This seems like a huge security issue, please prioritize this issue.

**Comment by sneljeroen** [comment]
This issue is an extension of the following issue, and should have the same solution
#112

**Comment by codewithtyler** [comment]
@sneljeroen I agree that this is an extension of issue #112. That said we will be tracking it over in that issue so that everyone will be able to get updates all at the same time.
